File: api/src/entities/WorklogSubmissionResult.ts
import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  CreateDateColumn,
  UpdateDateColumn,
  ManyToOne,
  JoinColumn,
} from 'typeorm';

import WorklogSubmission from './WorklogSubmission';

export type TargetSystem = 'TEMPO' | 'JIRA';
export type TargetResultStatus = 'PENDING' | 'SUCCESS' | 'FAILED';

@Entity('worklog_submission_result')
export default class WorklogSubmissionResult {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'submission_id' })
  submissionId: number;

  @ManyToOne(() => WorklogSubmission, (sub) => sub.results, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'submission_id' })
  submission: WorklogSubmission;

  @Column({ name: 'target_system', type: 'varchar', length: 10 })
  targetSystem: TargetSystem;

  @Column({ type: 'varchar', length: 10 })
  status: TargetResultStatus;

  @Column({ name: 'external_id', type: 'varchar', nullable: true })
  externalId: string | null;

  @Column({ name: 'request_payload', type: 'text' })
  requestPayload: string;

  @Column({ name: 'response_payload', type: 'text', nullable: true })
  responsePayload: string | null;

  @Column({ name: 'error_code', type: 'varchar', nullable: true })
  errorCode: string | null;

  @Column({ name: 'error_message', type: 'text', nullable: true })
  errorMessage: string | null;

  @CreateDateColumn({ name: 'created_at' })
  createdAt: Date;

  @UpdateDateColumn({ name: 'updated_at' })
  updatedAt: Date;
}

File: packages/planny-core/src/planny_core/models/worklog_submission_result.py
"""WorklogSubmissionResult model — mirrors ``api/src/entities/WorklogSubmissionResult.ts``.

Table name: ``worklog_submission_result``
"""

from __future__ import annotations

from datetime import datetime
from typing import TYPE_CHECKING

from sqlalchemy import DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from planny_core.database import Base

if TYPE_CHECKING:
    from planny_core.models.worklog_submission import WorklogSubmission


class WorklogSubmissionResult(Base):
    __tablename__ = "worklog_submission_result"

    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
    submission_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("worklog_submission.id", ondelete="CASCADE"),
    )

    target_system: Mapped[str] = mapped_column(String(10))
    status: Mapped[str] = mapped_column(String(10))
    external_id: Mapped[str | None] = mapped_column(String(255), nullable=True)
    request_payload: Mapped[str] = mapped_column(Text)
    response_payload: Mapped[str | None] = mapped_column(Text, nullable=True)
    error_code: Mapped[str | None] = mapped_column(String(255), nullable=True)
    error_message: Mapped[str | None] = mapped_column(Text, nullable=True)

    created_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime,
        default=datetime.utcnow,
        onupdate=datetime.utcnow,
    )

    # ── Relationships ──────────────────────────────────────────────────────
    submission: Mapped[WorklogSubmission] = relationship(
        "WorklogSubmission",
        back_populates="results",
    )

    def __repr__(self) -> str:
        return f"<WorklogSubmissionResult id={self.id}>"
